### PR TITLE
Control when path has not 'src' in it

### PR DESCRIPTION
**Pull request by gwydionmv**
Fixing issue #196 

An exception was being thrown when trying to substring a path not including 'src' in it. Now it's controlled that the index for the substring is valid before doing it.

### CODE REVIEW

**Review by hbelmiro — COMMENTED** [review]
@gwydionmv do you mind including a test for this use case?

**Comment by gwydionmv** [comment]
> I'm not exactly sure how I can manually force a Path for the execution of the plugin. I faced this issue because another plugin we are using in the project is setting it up internally

**Comment by hbelmiro** [comment]
@gwydionmv can't you just define a path without src with the property?

quarkus.openapi-generator.codegen.input-base-dir=openapi-definitions

**Comment by gwydionmv** [comment]
@hbelmiro the property it is relative to the project base directory. E.g. if you have a project called `MyProject` and decide to place the openapi file in `MyProject/openapi-definitions`, the property as you wrote it is valid.

The lack of src in the path is not related to the property, but to the sourceDir that's being set up automatically by maven and the quarkus compiler plugin. It's something similar to the issue we opened last week for the /test directory: it sets different paths depending on I don't know what yet. For your example tests, it adds the /test; for some of our projects, only /src/main; for others, it also includes /target/generated-sources...

To sum up, it's not something we can configure via input-base-dir property.

**Review by hbelmiro — APPROVED** [review]
LGTM.
Thanks for the contribution @gwydionmv.

**Comment by gwydionmv** [comment]
@ricardozanini please check this out when you have a moment to spare :)

**Review by ricardozanini — APPROVED** [review]
I'll release a patch version to include this fix. Thank you, @gwydionmv
